python_scripts/trees_hyperparameters.py

```diff
@@ -114,13 +114,23 @@ def plot_regression(model, X, y, ax=None):
 # %%
 plot_classification(tree_clf, data_clf[data_clf_columns],
                     data_clf[target_clf_column])
+accuracy = tree_clf.score(data_clf[data_clf_columns],
+                          data_clf[target_clf_column])
+
 plt.legend(bbox_to_anchor=(1.05, 1), loc='upper left')
-_ = plt.title(f"Shallow classification tree with max-depth of {max_depth}")
+_ = plt.title(
+    f"Shallow classification tree with max-depth of {max_depth}"
+    f"\n Accuracy of the fit: {accuracy:.2f}"
+)
```

Comment:
I don't see the point of measuring the scores only the training set. Here we speak about hyper-parameter tuning so this would be confusing to only display the training score. I think this notebook needs to be reworked to do a train test split and the plots should display both training and test errors, or neither.

Maybe the plots should be duplicated to each do 2 subplots: one with the prediction function displayed on top of a scatter plot of the samples of the training set (with the training score in the title) and another with the same prediction function displayed on top of a scatter plot of the samples of the testing set (with the testing score in the title).

And then we should comment on those scores to summarize the impact of the hyper-parameters in terms of the overfitting / underfitting trade-off.

Reply:
I split the PR.